### PR TITLE
[DEV-2625] WalletConnect multichain support

### PR DESCRIPTION
**Pull request by lapatric**

### CODE REVIEW

File: src/hooks/wallets/wallet-connect.hook.ts [hunk 1]
```diff
@@ -37,23 +49,42 @@ export interface DeepWallet {
...
+    type: 'function',
+    name: 'transfer',
+    stateMutability: 'nonpayable',
+    inputs: [
+      { name: 'recipient', type: 'address' },
+      { name: 'amount', type: 'uint256' },
+    ],
+    outputs: [{ type: 'bool' }],
+  },
+] as const;
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.hook.ts:71]: Can you use the ABI from src/static/erc20.abi.json?

File: src/hooks/wallets/wallet-connect.hook.ts [hunk 2]
```diff
@@ -78,140 +109,95 @@ export function useWalletConnect(): WalletConnectInterface {
...
+    return { asset, amount: Number(formatUnits(balance.value, balance.decimals)) };
+  }
+
+  async function requestChangeToBlockchain(blockchain?: Blockchain): Promise<void> {
+    if (!blockchain) return;
+    const chainId = Number(toChainId(blockchain));
+    if (getAccount(config).chainId !== chainId) {
+      await switchChain(config, { chainId: chainId as any, connector: getWalletConnectConnector() });
+      await new Promise((resolve) => setTimeout(resolve, 5000));
     }
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.hook.ts:162]: Why this timeout of 5s?

Comment (lapatric) [reply src/hooks/wallets/wallet-connect.hook.ts]: This was to allow for the wallet to complete the chain switch before the transaction confirmation prompt is sent but I removed it for now as it doesn't really make a bit difference.

File: src/hooks/wallets/wallet-connect.hook.ts [hunk 3]
```diff
@@ -37,23 +51,42 @@ export interface DeepWallet {
...
+    type: 'function',
+    name: 'transfer',
+    stateMutability: 'nonpayable',
+    inputs: [
+      { name: 'recipient', type: 'address' },
+      { name: 'amount', type: 'uint256' },
+    ],
+    outputs: [{ type: 'bool' }],
+  },
+] as const;
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.hook.ts:73]: Remove variable

File: src/hooks/wallets/wallet-connect.hook.ts [hunk 4]
```diff
@@ -109,87 +111,129 @@ export function useWalletConnect(): WalletConnectInterface {
...
+
+    baseError.walk((e) => e instanceof Error);
+    finalError = baseError;
 
-      return await writeContract(config, request);
+    if (finalError instanceof BaseError || finalError instanceof Error) {
+      throw new Error((finalError as BaseError).shortMessage);
+    } else {
+      throw new Error('An unexpected error occurred.');
     }
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.hook.ts:237]: Crazy code 😆

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**